### PR TITLE
I love timezones so much it makes me cry

### PR DESCRIPTION
**Pull request by BjornTwachtmann**

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/BlogsService/Repository/PostRepository.php [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 use App\BlogsService\Query\IsiteQuery\GuidQuery;
 use App\BlogsService\Query\IsiteQuery\SearchQuery;
+use App\Helper\ApplicationTimeProvider;
```
Comment (Tasiobg) [inline src/BlogsService/Repository/PostRepository.php]: This could have been commited by mistake

File: src/Helper/ApplicationTimeProvider.php [hunk 2]
```diff
@@ -5,17 +5,69 @@
 
 use Cake\Chronos\Chronos;
 use Cake\Chronos\Date;
+use DateTimeZone;
 
 class ApplicationTimeProvider
 {
-    public static function setTestDateTime(Chronos $time)
+    /** @var Chronos|null  */
+    private static $appTime = null;
```
Comment (Tasiobg) [inline src/Helper/ApplicationTimeProvider.php]: Class property default value is null, this is not required, you could break the internet please be more careful

File: src/Helper/ApplicationTimeProvider.php [hunk 3]
```diff
@@ -5,17 +5,69 @@
...
+            self::setLocalTimeZone();
+        }
+        return self::$localTimeZone;
+    }
+
+    public static function setDateTime(Chronos $time = null)
+    {
+        static::$localTime = null;
+        if ($time === null) {
+            static::$appTime = $time = Chronos::now();
```
Comment (Tasiobg) [inline src/Helper/ApplicationTimeProvider.php]: Is assining this to `$time` a coding style or does is have any usage or meaning? just wondering, doesn't need to be changed

**Pull request merged**